src/classes/RD2_ApiService_TEST.cls

```diff
@@ -246,6 +247,116 @@ private class RD2_ApiService_TEST {
+            UTIL_UnitTestData_TEST.mockId(npe03__Recurring_Donation__c.SObjectType));
+
+        System.assertEquals(0, installmentsByRd.size(),
+            'There should be no recurring donation installments returned when RD Id is invalid');
+    }
+
+    /**
+     * @description Validate an empty installment response when empty RD Id set is passed in
+     */
+    @IsTest
+    private static void shouldReturnNoInstallmentsWithEmptyRDIdSet() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Map<Id, List<Installment>> installmentsByRd = retrieveInstallmentsUsingApi(
+            Date.today(),
+            Date.today().addYears(2),
+            null);
+
+        System.assertEquals(0, installmentsByRd.size(),
+            'There should be no recurring donation installments returned when RD Id Set is empty');
+    }
+
+    /**
+     * @description Validate a valid installment response when start date is not passed in
+     */
+    @IsTest
+    private static void shouldReturnValidInstallmentsWithMissingStartDate() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+        npe03__Recurring_Donation__c rd = rdGateway.getRecords()[0];
+
+        Map<Id, List<Installment>> installmentsByRd = retrieveInstallmentsUsingApi(
+            null,
+            date.today().addYears(2),
+            rd.Id);
+
+        System.assertNotEquals(0, installmentsByRd.size(),
+            'Recurring donation installments should be returned when Start Date is missing');
+    }
+
+    /**
+     * @description Validate an empty installment response when End Date is missing
+     */
+    @IsTest
+    private static void shouldReturnNoInstallmentsWithMissingEndDate() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+        npe03__Recurring_Donation__c rd = rdGateway.getRecords()[0];
+
+        Map<Id, List<Installment>> installmentsByRd = retrieveInstallmentsUsingApi(
+            date.today(),
+            null,
+            rd.Id);
+
+        System.assertNotEquals(0, installmentsByRd.size(),
+            'There should be no recurring donation installments returned when End Date is missing');
+    }
+
+    /**
+     * @description Validate a valid installment response when valid parameters passed in
+     */
+    @IsTest
+    private static void shouldReturnValidInstallmentsWithValidParameters() {
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+        npe03__Recurring_Donation__c rd = rdGateway.getRecords()[0];
+
+        Map<Id, List<Installment>> installmentsByRd = retrieveInstallmentsUsingApi(
+            date.today(),
+            date.today().addYears(2),
+            rd.Id);
+        System.assertEquals(25, installmentsByRd.get(rd.Id).size(),
+            '25 Recurring donation installments should be returned when there is no pause schedule');
+
+        RD2_ScheduleService_TEST.createPauseSchedule(rd.Id, Date.today().addMonths(1), Date.today().addMonths(2));
```

Comment:
Let's create two tests out of this one since there are two concepts tested: installments when params are valid and installments when the RD is paused for two installments.
Let's store 25 and 23 as vars so it is easier to know what these numbers represent.